src/types/index.ts

```diff
@@ -10,15 +10,15 @@ export type TriggerFunction = (persist: () => void) => TriggerUninstallFunction;
 
 export type PersistedData<T> = T | string | null;
 
-export interface PersistentStorage<T> {
-  getItem: (key: string) => Promise<T | null> | T | null;
-  setItem: (key: string, data: T) => Promise<T> | Promise<void> | void | T;
-  removeItem: (key: string) => Promise<T> | Promise<void> | void;
+export interface PersistentStorage {
```

Comment:
The definitions within `PersistentStorage` were updated to match TypeScript's internal types for the `Storage` interface, but then amended to also support asynchronous storage APIs too, as needed by AsyncStorage, `localForage` and Ionic Storage.